include/appimage/appimage.h

```diff
@@ -37,6 +37,17 @@ void appimage_extract_file_following_symlinks(const char* appimage_file_path, co
  * https://specifications.freedesktop.org/thumbnail-spec/0.8.0/index.html
  */
 void appimage_create_thumbnail(const char* appimage_file_path);
+
+/* List files contained in the AppImage file.
+ * Returns: a newly allocated char** ended at NULL. If no files ware found also is returned a {NULL}
+ *
+ * You should ALWAYS take care of releasing this structure like this:
+ * char **list;
+ * for (char **ptr = list; *ptr != null; ptr ++)
+ *   free(*ptr)
+ * free(list);
```

Comment:
Argh. This seems really complicated. Can we add a function that frees the structure? For example, `appimage_list_free(void**)`?

Reply:
That's an simple strings array in C, not a complex data structure.

I think that every C programmer can handle it. Also I don't like to put too many stuff in the APIs but if you think that it will be helpful then yes we can add it.

Reply:
@azubieta as soon as you need to provide a "code template" to free a data structure, it's complex enough to deserve its own function to be freed.